packages/material-ui/src/AppBar/AppBar.js

```diff
@@ -29,71 +58,88 @@ export const styles = (theme) => {
         // Prevent the app bar to be visible on each printed page.
         position: 'absolute',
       },
-    },
+    }),
     /* Styles applied to the root element if `position="absolute"`. */
-    positionAbsolute: {
+    ...(styleProps.position === 'absolute' && {
       position: 'absolute',
       zIndex: theme.zIndex.appBar,
       top: 0,
       left: 'auto',
       right: 0,
-    },
+    }),
     /* Styles applied to the root element if `position="sticky"`. */
-    positionSticky: {
+    ...(styleProps.position === 'sticky' && {
       // ⚠️ sticky is not supported by IE11.
       position: 'sticky',
       zIndex: theme.zIndex.appBar,
       top: 0,
       left: 'auto',
       right: 0,
-    },
+    }),
     /* Styles applied to the root element if `position="static"`. */
-    positionStatic: {
+    ...(styleProps.position === 'static' && {
       position: 'static',
-    },
+    }),
     /* Styles applied to the root element if `position="relative"`. */
-    positionRelative: {
+    ...(styleProps.position === 'relative' && {
       position: 'relative',
-    },
+    }),
     /* Styles applied to the root element if `color="default"`. */
-    colorDefault: {
+    ...(styleProps.color === 'default' && {
       backgroundColor: backgroundColorDefault,
       color: theme.palette.getContrastText(backgroundColorDefault),
-    },
+    }),
     /* Styles applied to the root element if `color="primary"`. */
-    colorPrimary: {
+    ...(styleProps.color === 'primary' && {
       backgroundColor: theme.palette.primary.main,
       color: theme.palette.primary.contrastText,
-    },
+    }),
     /* Styles applied to the root element if `color="secondary"`. */
-    colorSecondary: {
+    ...(styleProps.color === 'secondary' && {
       backgroundColor: theme.palette.secondary.main,
       color: theme.palette.secondary.contrastText,
-    },
+    }),
     /* Styles applied to the root element if `color="inherit"`. */
-    colorInherit: {
+    ...(styleProps.color === 'inherit' && {
       color: 'inherit',
-    },
+    }),
     /* Styles applied to the root element if `color="transparent"`. */
-    colorTransparent: {
+    ...(styleProps.color === 'transparent' && {
       backgroundColor: 'transparent',
       color: 'inherit',
-    },
+    }),
   };
-};
+});
 
-const AppBar = React.forwardRef(function AppBar(props, ref) {
-  const { classes, className, color = 'primary', position = 'fixed', ...other } = props;
+const AppBar = React.forwardRef(function AppBar(inProps, ref) {
+  const {
+    className,
+    color = 'primary',
+    position = 'fixed',
+    classes: inClasses,
+    ...other
+  } = useThemeProps({
+    props: inProps,
+    name: 'MuiAppBar',
+  });
+
+  const styleProps = {
+    ...other,
+    color,
+    position,
+    classes: inClasses,
```

Comment:
```suggestion
```

Reply:
Why I made inClasses because 
![image](https://user-images.githubusercontent.com/8037810/104778391-f3f33000-5785-11eb-85d6-cc4e1e52a545.png)

Reply:
Yes, we have issue https://github.com/mui-org/material-ui/issues/24373 I was planning to add the `classes` in the list of props we are not forwarding to the DOM elements. That way it will be solved for all components. But feel free to leave it if you want, I will do a cleanup in all components when I work on it

Reply:
I can remove it because your fix solves a global problem I don't mind.